### PR TITLE
[docs] Disallow object assign explanation

### PR DESCRIPTION
**Pull request by sharnik**
> Explain the **motivation** for making this change. What existing problem does the pull request solve?

Provides better documentation for a confusing (and common) error message. Solves https://github.com/facebook/react-native/issues/3634

> **Test plan (required)**

This is just an error message shown in dev environment. I ran the tests and they pass.

### CODE REVIEW

**Comment by ide** [comment]
I find the old error message clearer, can you combine the two? Really the problem is that we found enumerable properties on the prototype chain and won't copy them..

**Comment by sharnik** [comment]
Sure, no problem. Done.

**Comment by ide** [comment]
@facebook-github-bot shipit

File: packager/react-packager/src/Resolver/polyfills/polyfills.js [hunk 1]
```diff
@@ -54,8 +55,9 @@ Object.assign = function(target, sources) {
         var hasOwnProperty = Object.prototype.hasOwnProperty;
         if (!hasOwnProperty.call(nextSource, key)) {
           throw new TypeError(
-            'One of the sources for assign has an enumerable key on the ' +
-            'prototype chain. This is an edge case that we do not support. ' +
+            'One of the sources for assign has an enumerable key on the' +
```
Comment (sharnik) [inline packager/react-packager/src/Resolver/polyfills/polyfills.js]: Fixed, thanks!

**Comment by sharnik** [comment]
@davidaurelio  Any ideas on those test failures? Is there anything I should do?

The logs are too long for me to read and before adding the single space they passed.

**Comment by hramos** [comment]
Giving this another try.

**Comment by NoelBroda** [comment]
So, what about this PR?

**Comment by sharnik** [comment]
It's merged to master and closed.

**Comment by kevin91liu** [comment]
Are there any plans to make it spec compliant? I am trying to send a file via FormData and this it seems that this non-spec-compliance is blocking me from doing so. Does anyone have a good workaround?

**Pull request closed without merging**